praveenkrishna0512 iP

praveenkrishna0512 opened the pull request:
## Duke!!

> “Stop wasting time remembering tasks man ” – Probably some guy


Duke!! reduces your cognitive load because it's:

- text-based
- easy to learn
- FAST SUPER FAST to use


Accelerate your life by doing the following:
1. download it from [here](https://github.com/praveenkrishna0512/ip/releases).
2. run the program by double clicking it.
3. add your tasks through the CLI.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:
- [X] Managing tasks
- [X]  Managing deadlines
- [ ]  Reminders (coming soon)

Use the `todo` command to add tasks that you have to do, with no deadline!
`todo <name-of-task>`

Here is a free function, just for the fun of it.
```java
public class Main {
    public static void main(String[] args) {
        System.out.println("You are great");
    }
}
```

Commit:
Add Gradle support

praveenkrishna0512 committed:
edited gitignore

praveenkrishna0512 committed:
Completed Level 1 - Greet, Echo and Exit

praveenkrishna0512 committed:
changed the prompts slightly to make it more user friendly

praveenkrishna0512 committed:
added add to list functionality

praveenkrishna0512 committed:
added print list functionality

praveenkrishna0512 committed:
Curated Task class

praveenkrishna0512 committed:
added getters/setters to Task.java + CommandHandler functionality to Duke.java

praveenkrishna0512 committed:
added mark and unmark functionality

praveenkrishna0512 committed:
Added subclasses of Task (ToDo, Event, Deadline)

praveenkrishna0512 committed:
Added toString to all subclasses of Task

praveenkrishna0512 committed:
added functionality for tracking three types of tasks

praveenkrishna0512 committed:
Automated Text UI Testing

praveenkrishna0512 committed:
Made DukeException and IllegalCommandException classes:

praveenkrishna0512 committed:
Added exception handling

praveenkrishna0512 committed:
added delete functionality

praveenkrishna0512 committed:
tagging for A-Enums

praveenkrishna0512 committed:
Ensure that Duke.javs follows Java convention

praveenkrishna0512 committed:
Allow tasks to accept done as constructor parameter

praveenkrishna0512 committed:
Enable functionality to load tasks from memory

praveenkrishna0512 committed:
Add writeToMemory functions for all Task subclasses

praveenkrishna0512 committed:
Add functionality to save tasks

praveenkrishna0512 committed:
Update .gitignore file

praveenkrishna0512 committed:
Configure Deadline and Event to accept LocalDateTime

praveenkrishna0512 committed:
Allow deadline and event commands to accept only dates

praveenkrishna0512 committed:
Merge branch 'branch-Level-7'

praveenkrishna0512 committed:
Merge branch-Level-8 to master after resolving conflicts

praveenkrishna0512 committed:
Create Storage, Ui and TaskList class files

praveenkrishna0512 committed:
Add some functionality for Storage class

praveenkrishna0512 committed:
Add some functionality for TaskList class

praveenkrishna0512 committed:
Add some functionality for Parser class

praveenkrishna0512 committed:
Add some functionality for Ui class

praveenkrishna0512 committed:
Add MemoryFailedException

praveenkrishna0512 committed:
Configure Duke to follow abstractions made previously

praveenkrishna0512 committed:
Ensure Duke shows logo on startup

praveenkrishna0512 committed:
Organize classes into packages

praveenkrishna0512 committed:
Merge branch 'add-gradle-support' of https://github.com/praveenkrishna0512/ip into add-gradle-support

praveenkrishna0512 committed:
Set up checkstyle configuration

praveenkrishna0512 committed:
Configure checkstyle.xml

praveenkrishna0512 committed:
Add functionality for gradles to build and run Duke

praveenkrishna0512 committed:
Configure JUnit tests for Deadline, Event and TaskList classes

praveenkrishna0512 committed:
Add JavaDoc header comments within the Task class and its subclasses

praveenkrishna0512 committed:
Add JavaDoc header comments for classes within utils package

praveenkrishna0512 committed:
Resolve unused imports warning

praveenkrishna0512 committed:
Amend test classes to abide by import conventions

praveenkrishna0512 committed:
Amend TaskList to follow JavaDoc convention

praveenkrishna0512 committed:
Allow parser to recognized find command

praveenkrishna0512 committed:
Add functionality to search for tasks with keyword

praveenkrishna0512 committed:
Merge branch 'branch-A-JavaDoc'

praveenkrishna0512 committed:
Fix merge conflicts between branch-Level-9 and master

daitenshionyan commented on src/main/java/duke/utils/Parser.java:
```diff
@@ -0,0 +1,114 @@
+                DateTimeFormatter dateFormat = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
+                LocalDateTime start = LocalDateTime.parse(startString, dateFormat);
+                LocalDateTime end = LocalDateTime.parse(endString, dateFormat);
+                // TODO: Check if start date is after end date
```
I like this remider comment.

praveenkrishna0512 replied:
Thank you!

daitenshionyan commented on src/test/java/duke/tasks/DeadlineTest.java:
```diff
@@ -0,0 +1,24 @@
+    private Deadline deadline;
+
+    @Test
+    @DisplayName("The toString method in the Deadline class should return a string of specified format.")
```
I like how there are display names for tests!

daitenshionyan commented on src/main/java/duke/tasks/Event.java:
```diff
@@ -0,0 +1,49 @@
+     *
+     * @return String representation of the Event, for storage in memory.
+     */
+    public String convertToMemoryString() {
```
I like how the names are not shortened

daitenshionyan commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,54 @@
+    /**
+     * Runs the Duke application.
+     */
+    public void run() {
+        Ui.printOnStartup();
+        boolean promptAgain = true;
+        while (promptAgain) {
+            Ui.printPrompt();
+            String command = Ui.listen();
+            try {
+                promptAgain = Parser.handleCommands(command, this.allTasks);
+            } catch (DukeException e) {
+                Ui.println(e.toString());
+            }
+            Ui.printDottedLine();
+        }
+        this.storage.saveTasks(this.allTasks);
+    }
```
I like how you extracted out the run logic!!

daitenshionyan commented on src/main/java/duke/utils/Parser.java:
```diff
@@ -0,0 +1,114 @@
+        String command = rawCommand;
+        String arguments = "";
+        if (commandIndex != -1) {
+            // There is a space character in the command
```
I like how this comment indicates what the if is checking for.

daitenshionyan commented on src/main/java/duke/utils/Storage.java:
```diff
@@ -0,0 +1,103 @@
+     */
+    public void loadTasks(TaskList allTasks) throws MemoryFailedException {
+        try {
+            boolean success = this.memory.createNewFile();
```
Perhaps an even more boolean name like `isSuccess`? I feel `success` is sufficient but maybe it could be even clearer although the variable is only used within this method.

praveenkrishna0512 replied:
You are right, thank you for the suggestion!

daitenshionyan reviewed (commented):
I love your consistent new line spacings it seems to really aids in readability. I really like your documentations as well!!

harin0826 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,54 @@
+     */
+    public void run() {
+        Ui.printOnStartup();
+        boolean promptAgain = true;
```
Could this boolean be named to shouldPromptAgain?

praveenkrishna0512 replied:
That's true, thank you!

harin0826 reviewed (commented):
I learnt a lot from your code! Small coding standard issues which I believe you have already fixed by now, thank you! :)

praveenkrishna0512 committed:
Configure start method in Duke.java

praveenkrishna0512 committed:
Ensure that base GUI compiles and appears on screen

praveenkrishna0512 committed:
Set up the GUI for Duke

praveenkrishna0512 committed:
Remove unused import statements

praveenkrishna0512 committed:
Remove unused import statements

praveenkrishna0512 committed:
Merge branch 'branch-Level-10'

praveenkrishna0512 committed:
Improve code quality

praveenkrishna0512 committed:
Add assertions to ensure code correctness

Assumptions about the state of the program at any point
during program execution should be ensured to hold true.

Let's use assertions to help us achieve this. We ensure
that the following behaviours:
* Resulting response string from a valid command should
not be empty.
* Memory Path Array supplied to the Storage constructor
should not be empty.
* Number of results arising from a search query cannot
be negative.
* Response string obtained by Duke from the Parser cannot
be null

Using assertions liberally is reccommended to reduce the
occurences of bugs due to human error.

praveenkrishna0512 committed:
Add assertions to ensure code correctness

Assumptions about the state of the program at any point
during program execution should be ensured to hold true.

Let's use assertions to help us achieve this. We ensure
that the following behaviours:
* Resulting response string from a valid command should
not be empty.
* Memory Path Array supplied to the Storage constructor
should not be empty.
* Number of results arising from a search query cannot
be negative.
* Response string obtained by Duke from the Parser cannot
be null

Using assertions liberally is reccommended to reduce the occurences of bugs due to human error.

praveenkrishna0512 committed:
Apply SLAP to handleCommands function in Parser

The Single Layer of Abstraction Principle (SLAP) allows us
to write neat code that is easily readable.

Let's use SLAP to tidy up the function body of
handleCommands within the Parser class. We abstract the
process of handling each command type into its own function.

We reccommend using SLAP to ensure easy readability and
ease of debugging.

praveenkrishna0512 committed:
Merge branch 'A-Assertions' of https://github.com/praveenkrishna0512/ip into A-Assertions

praveenkrishna0512 committed:
Merge pull request #2 from praveenkrishna0512/A-Assertions

Add assertions to ensure code correctness

praveenkrishna0512 committed:
Merge branch 'master' into A-CodeQuality

praveenkrishna0512 committed:
Merge pull request #1 from praveenkrishna0512/A-CodeQuality

Apply SLAP to handleCommands function in Parser

praveenkrishna0512 committed:
Compile Duke's responses into one message

praveenkrishna0512 committed:
Allow for friendlier syntax

praveenkrishna0512 committed:
Merge pull request #3 from praveenkrishna0512/branch-BCD-Extension

Allow for friendlier syntax

praveenkrishna0512 committed:
Ensure code follows CS2103T standards

praveenkrishna0512 committed:
Amend GUI to show the entire reply text from Duke

praveenkrishna0512 committed:
Add title to application window

praveenkrishna0512 committed:
Update README.md with instructions for Add Task

praveenkrishna0512 committed:
Update README.md with complete Features section

praveenkrishna0512 committed:
Update title of README.md
